modules/apim/manifests/init.pp

```diff
@@ -1,17 +1,47 @@
+# ----------------------------------------------------------------------------
+#  Copyright (c) 2018 WSO2, Inc. http://www.wso2.org
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+# ----------------------------------------------------------------------------
+
+# Class: apim
+# Init class of API Manager default profile
 class apim (
-  $user                 = $apim::params::user,
-  $user_group           = $apim::params::user_group,
-  $user_group_id        = $apim::params::user_group_id,
-  $service_name         = $apim::params::service_name,
+  $user                  = $apim::params::user,
+  $user_id               = $apim::params::user_id,
+  $user_group            = $apim::params::user_group,
+  $user_group_id         = $apim::params::user_group_id,
+  $service_name          = $apim::params::service_name,
+  $template_list         = $apim::params::template_list,
+  $jre_version           = $apim::params::jre_version,
```

Comment:
IMO since JAVA_HOME can be constructed as `$install_path/jre/$jre_version` and it is only used in `wso2server.sh` erb template I think this can remain like this. 
WDYT? 
But it's better if we can come to a common way of doing this in ansible and puppet.

Reply:
Having the jre_version is redundant since it is not used in any other places. We could directly construct java_home as follows.
JAVA_HOME=$install_path/jre/<jre_version>
The purpose of having jre_version as a separate parameter is to use it somewhere else to construct another parameter or use it as it is. Since it is not the case, my opinion is to remove jre_version.

Reply:
This cannot be constructed before knowing the correct $install_path of the installer.
eg: in Ubuntu install_path begin with `usr/lib/...` and in CentOs it begins with `/usr/lib64/...`
So without knowing it, we cannot create the correct JAVA_HOME. We first have to check the os family and decide it. So that is why in puppet we had to construct it inside template as 
`JAVA_HOME="<%=@install_path%>/jre/<%=@jre_version%>"`